packages/transactions/src/authorization.ts

```diff
@@ -488,11 +517,77 @@ function verifySingleSig(
     condition.signature
   );
 
-  // TODO: verify pub key
+  // address version arg doesn't matter for signer hash generation
+  const addrBytes = addressFromPublicKeys(0, condition.hashMode, 1, [pubKey]).hash160;
+
+  if (addrBytes !== condition.signer)
+    throw new VerificationError(
+      `Signer hash does not equal hash of public key(s): ${addrBytes} != ${condition.signer}`
+    );
 
   return nextSigHash;
 }
 
+function verifyMultiSig(
+  condition: MultiSigSpendingConditionOpts,
+  initialSigHash: string,
+  authType: AuthType
+): string {
+  const publicKeys: StacksPublicKey[] = [];
+  let curSigHash = initialSigHash;
+  let haveUncompressed = false;
+  const numSigs = new Uint16Array(1);
```

Comment:
@CharlieC3 I think Faried used it to save space, number takes 2^53, so he used `Uint16Array`, because I don't think there is a built-in `Uint16` type supported in the typescript. 

I can change it to the `number` if you want?

Reply:
To be honest, I don't remember why I went with this approach because I haven't seen it used elsewhere in this repository, or used it myself somewhere.  Maybe I was trying to make it explicit that it was a 16-bit value (the Rust code uses a `u16`).

I don't think I was trying to save space by using a single-element `Uint16Array`; it likely takes up more space than a normal variable.